### PR TITLE
Refactor pollForTentativeDevice

### PR DESCRIPTION
**Pull request by nmattia**
This updates the `pollForTentativeDevice` screen to be more consistent with other pages, and includes the following fixes:
* The `poll` function loop is made explicit (also meaning the function won't be nested indefinitely anymore)
* The user flow logic (which screens to show, error handling, etc) is moved out of `pollForTentativeDevice` and is instead handled in a new function, `addRemoteDevice` mirroring `addLocalDevice` (meaning the only concern of `pollForTentativeDevice` is to poll for tentative device)
* All promise results are changed to async/await instead of `.then()`
* The `AsyncCountdown` is used, meaning less state is shared

### CODE REVIEW

File: src/frontend/src/flows/addDevice/manage/addRemoteDevice.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+// The flow for adding a remote (i.e. other browser, non-hardware) device
+export const addRemoteDevice = async ({
+  userNumber,
+  connection,
+}: {
+  userNumber: bigint;
+  connection: AuthenticatedConnection;
+}): Promise<void> => {
+  // Enter registration mode and get info about the anchor
+  const [timestamp, anchorInfo] = await withLoader(() =>
```
Comment (peterpeterparker) [inline src/frontend/src/flows/addDevice/manage/addRemoteDevice.ts:16]: 1. the function use to be `async`, not needed anymore?

```
await withLoader(withLoader(async () => 
```

2. nth: even if deferred I would not be against specifying the return type for readability

```
const [timestamp, anchorInfo] = await withLoader((): Promise<[something, something]> => ...
```

File: src/frontend/src/flows/addDevice/manage/addRemoteDevice.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    if (result === "timeout") {
+      // On timeout, notify the user
+      await displayError({
+        title: "Timeout Reached",
+        message:
+          'The timeout has been reached. For security reasons the "add device" process has been aborted.',
+        primaryButton: "Ok",
+      });
+      return;
```
Comment (peterpeterparker) [inline src/frontend/src/flows/addDevice/manage/addRemoteDevice.ts:40]: can result be here something else that timeout or that's an edge case? can you maybe add a comment next to this return to explains what happens if we land here?

File: src/frontend/src/flows/addDevice/manage/pollForTentativeDevice.ts [hunk 3]
```diff
@@ -55,110 +67,64 @@ const pageContent = (userNumber: bigint) => {
...
-    await init(userNumber, connection, timestamp);
-  }
-};
+  // Poll repeatedly
+  const polling = poll(userNumber, connection, () => countdown.hasStopped());
 
-export const renderPollForTentativeDevicePage = (userNumber: bigint): void => {
-  const container = document.getElementById("pageContent") as HTMLElement;
-  render(pageContent(userNumber), container);
+  return await Promise.race([page, polling]);
```
Comment (peterpeterparker) [inline src/frontend/src/flows/addDevice/manage/pollForTentativeDevice.ts:106]: the `await` is useful here?

File: src/frontend/src/flows/addDevice/manage/pollForTentativeDevice.ts [hunk 4]
```diff
@@ -55,110 +67,64 @@ const pageContent = (userNumber: bigint) => {
...
     if (shouldStop()) {
-      return null;
+      return "timeout";
     }
-    const tentativeDevice = getTentativeDevice(response);
-    if (tentativeDevice) {
+
+    const anchorInfo = await connection.getAnchorInfo();
+    const tentativeDevice =
+      anchorInfo.device_registration[0]?.tentative_device[0];
```
Comment (peterpeterparker) [inline src/frontend/src/flows/addDevice/manage/pollForTentativeDevice.ts:121]: this - `anchorInfo.device_registration[0]?.tentative_device[0]` - appears twice in this PR, maybe worth a utils?

**Review by peterpeterparker — APPROVED** [review]
LGTM, minor comments. Thanks

File: src/frontend/src/flows/addDevice/manage/addRemoteDevice.ts (hunk 1 above)
Comment (nmattia) [reply src/frontend/src/flows/addDevice/manage/addRemoteDevice.ts:16]: > the function use to be async, not needed anymore?

Not needed; wasn't needed before either as far as I could tell

> nth: even if deferred I would not be against specifying the return type for readability

Sure, I'll actually add it to the LHS instead if ok for you (i.e.` const [ ... ] : [Timestamp, ...] =`)

File: src/frontend/src/flows/addDevice/manage/addRemoteDevice.ts (hunk 2 above)
Comment (nmattia) [reply src/frontend/src/flows/addDevice/manage/addRemoteDevice.ts:40]: The result can be a timeout, a cancellation or device information. For the two first cases (timeout, cancel) there's not much we can do. I'm adding a comment!

File: src/frontend/src/flows/addDevice/manage/pollForTentativeDevice.ts (hunk 3 above)
Comment (nmattia) [reply src/frontend/src/flows/addDevice/manage/pollForTentativeDevice.ts:106]: It's not! good catch

File: src/frontend/src/flows/addDevice/manage/pollForTentativeDevice.ts (hunk 4 above)
Comment (nmattia) [reply src/frontend/src/flows/addDevice/manage/pollForTentativeDevice.ts:121]: There was a helper around this but I found it more confusing than just inlining the definition; if you feel strongly I'm happy to re-add one!

File: src/frontend/src/flows/addDevice/manage/addRemoteDevice.ts (hunk 1 above)
Comment (nmattia) [reply src/frontend/src/flows/addDevice/manage/addRemoteDevice.ts]: Done!

**Review by frederikrothenberger — APPROVED** [review]
This is a nice refactoring 👍

**Pull request merged**